app/controllers/posts_controller.rb

```diff
@@ -9,4 +9,27 @@ def show
     @user = @post.author
     @comments = @post.comments
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @new_post = current_user.posts.new(post_params)
+    respond_to do |format|
+      format.html do
+        if @new_post.save
+          redirect_to "/users/#{@new_post.author.id}/posts/", notice: 'Success!'
+        else
+          render :new, alert: 'Error occured!'
+        end
+      end
+    end
+  end
+
+  private
```

Comment:
- [ ] currently when I try creating a post I get an error of unknow variable curren_user, kindly add a method current user in your application_controller.rb. it should look like this 
```
  class ApplicationController < ActionController::Base
    def current_user
      ......
      end
end
```
![Screenshot from 2022-06-08 13-55-53](https://user-images.githubusercontent.com/66097371/172622340-b98aab01-3b2e-4ff3-8342-bf4b3f117c7a.png)